### PR TITLE
Localization algebra

### PR DESCRIPTION
**Pull request by jpoiret**
Hi,

This is an attempt to define the localization of an algebra at an arbitrary multiplicatively closed subset of it, compared with the current localization at a multiplicatively closed subset of the base ring. I've had many attempts, trying to use CT and the fact that algebras over a base ring are equivalent to rings under a base ring, but the definitional behavior was really hard to predict and things just didn't compute well. I chose instead to just extend the existing construction in the most straightforward way. I'm still trying to formulate a recursion principle using the universal property, so that we're able to prove things about the localization without resorting to its actual construction.

This depends on a rebased version of #931.

LMKWYT

### CODE REVIEW

File: Cubical/Algebra/CommAlgebra/LocalisationAlgebra.agda [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+extending the ring construction mechanically to using the fact the category of
+algebras is a coslice category and exactly re-using the localization of rings.
+Here, we chose the former, because after trying out the latter for some time, it
+turned out that its computational behavior wasn't as nice as expected, and it
+made Agda struggle a lot.
+-}
+
+{-# OPTIONS --safe #-}
+
+open import Cubical.Core.Everything
```
Comment (felixwellen) [inline Cubical/Algebra/CommAlgebra/LocalisationAlgebra.agda:13]: Probably not needed.

File: Cubical/Algebra/CommAlgebra/LocalisationAlgebra.agda [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+    _⁻¹ : (a : ⟨ A ⟩) → ⦃ a ∈ _ˣ ⦄ → ⟨ A ⟩
+    _⁻¹ a ⦃ a-inv ⦄ = fst a-inv
+
+  open Units
+
+  open isMultClosedSubset SMultClosedSubset
+  private Aᵣ = CommAlgebra→CommRing A
+
+  private module S⁻¹ADef where
```
Comment (felixwellen) [inline Cubical/Algebra/CommAlgebra/LocalisationAlgebra.agda:60]: Maybe a more descriptive name.

Comment (felixwellen) [reply Cubical/Algebra/CommAlgebra/LocalisationAlgebra.agda:60]: Maybe "AlgebraStructureOnLocalization"

File: Cubical/Algebra/CommAlgebra/LocalisationAlgebra.agda [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+  S⁻¹AAsCommAlgebra : CommAlgebra R ℓAlg
+  S⁻¹AAsCommAlgebra = let open S⁻¹ADef
+    in commAlgebraFromCommRing {R = R} S⁻¹Aᵣ _⋆_ ·Assoc⋆ ⋆DistR+ ⋆DistL+ ⋆IdL ⋆AssocL
+
+  S⁻¹AAsCommAlgebra→CommRing : CommAlgebra→CommRing S⁻¹AAsCommAlgebra ≡ RLoc.S⁻¹RAsCommRing
+  S⁻¹AAsCommAlgebra→CommRing = let open S⁻¹ADef
+    in commAlgebraFromCommRing→CommRing {R = R} S⁻¹Aᵣ _⋆_ ·Assoc⋆ ⋆DistR+
+                                        ⋆DistL+ ⋆IdL ⋆AssocL
+
+  module S⁻¹AUnversalProp where
```
Comment (felixwellen) [inline Cubical/Algebra/CommAlgebra/LocalisationAlgebra.agda:133]: Typo "Universal"

**Review by felixwellen — CHANGES_REQUESTED** [review]
Discussed in person.

**Comment by felixwellen** [comment]
Checking the library took 47min in the CI, we should investigate.

**Comment by felixwellen** [comment]
Both master and this branch take 27min on my machine, so it should be fine.

File: Cubical/Algebra/CommAlgebra/LocalisationAlgebra.agda (hunk 1 above)
Comment (jpoiret) [reply Cubical/Algebra/CommAlgebra/LocalisationAlgebra.agda]: I've removed all of those unnecessary imports.

**Review by felixwellen — CHANGES_REQUESTED** [review]
LGTM - just a request for comment.

**Pull request merged**